File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {

stack<char> st;


        for (char c : s) {
            if (isalnum(c)) {
                st.push(tolower(c));
            }
        }

        
        for (char c : s) {
            if (isalnum(c)) {
                if (st.top() != tolower(c)) return false; 
                st.pop();
            }
        }

        return true;  
    }
};

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        res = ""

        for i in s:
            if i.isalnum():
                res += i.lower()
        rev_res = res[::-1]
        if res == rev_res:
            return True
        return False
        